lib/tasks/reports.rake

```diff
@@ -120,3 +120,96 @@ namespace :reports do
+                end
+      base = {
+        "host" => "host#{host}.example.com", "reported_at" => Time.now.utc.to_s,
+        "status" => {
+          "applied" => host,
+          "restarted" => host * 2,
+          "failed" => host * 3,
+          "failed_restarts" => 0,
+          "skipped" => 1,
+          "pending" => 9999999,
+        },
+        "metrics" => metrics,
+        "logs" => []
+      }
+      (1..logs).each do |i|
+        src = if is_ansible
+                "common : Copy default motd %s" % (i.even? ? randn : 0)
+              elsif is_puppet
+                "//Node/Puppet[%s]" % (i.even? ? randn : 0)
+              else
+                "A unique message %s" % (i.even? ? randn : 0)
+              end
+        msg = if is_ansible
+                "{\"_ansible_parsed\": true, \"group\": \"root\", \"uid\": 0, \"checksum\": \"%s\", \"changed\": false, \"owner\": \"root\", \"state\": \"file\", \"gid\": 0, \"mode\": \"0644\", \"diff\": {\"after\": {\"path\": \"/etc/motd\"}, \"before\": {\"path\": \"/etc/motd\"}}, \"invocation\": {\"module_args\": {\"directory_mode\": null, \"force\": false, \"remote_src\": null, \"path\": \"/etc/motd\", \"owner\": \"root\", \"follow\": false, \"group\": \"root\", \"unsafe_writes\": null, \"state\": \"file\", \"content\": null, \"serole\": null, \"diff_peek\": null, \"setype\": null, \"dest\": \"/etc/motd\", \"selevel\": null, \"original_basename\": \"motd.txt\", \"regexp\": null, \"validate\": null, \"src\": \"motd.txt\", \"seuser\": null, \"recurse\": false, \"delimiter\": null, \"mode\": null, \"attributes\": null, \"backup\": null}}, \"path\": \"/etc/motd\", \"size\": 1090, \"_ansible_no_log\": false}" % (i.even? ? Digest::SHA1.hexdigest(randn.to_s) : "0a381ff6a86081af6dc957a77c7e2017a3244c4c")
+              elsif is_puppet
+                "A Puppet message"
+              else
+                "A message"
+              end
+        base["logs"].append({
+                              "log" => { "sources" => { "source" => src },
+                              "messages" => { "message" => msg },
+                              "level" => (i.even? ? "notice" : "err") },
+                            })
+      end
+      base["reporter"] = "ansible" if is_ansible
+      base
+    end
+
+    User.as_anonymous_admin do
+      Rails.logger.level = Logger::ERROR
+      Foreman::Logging.logger('permissions').level = Logger::ERROR
+      Foreman::Logging.logger('audit').level = Logger::ERROR
+      origin = ENV['origin'] || 'Puppet'
+      hosts = ENV['hosts'] || 10
+      reports = ENV['reports'] || 50
+      lines = ENV['lines'] || 100
+      total_time = 0
+      (1..reports).each do |i|
+        host_id = i % hosts
+        report = make_report(origin, host_id + 1, lines)
+        start_time = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)
+        ConfigReport.import(report)
+        total_time += Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond) - start_time
```

Comment:
It's an old document, but https://guides.rubyonrails.org/v3.2.13/performance_testing.html suggests there is a framework built into Rails to do this kind of testing.

Reply:
Yeah but this is not a performance test, it's actually more of a test seeder (forgery?) which prints rate data as well. We do have the `telemetry_duration_histogram` method to do this via telemetry API, but this is really just a quick and dirty "print it to the console" measurement to keep me informed.